components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIGatewayPublisherImpl.java

```diff
@@ -114,6 +122,10 @@ public void deleteAPI(API api) throws GatewayException {
         if (log.isDebugEnabled()) {
             log.debug("API : " + api.getName() + " deleted event has been successfully published to broker");
         }
+        if (api.hasOwnGateway()) {
+            // Delete the Gateway - check how we can assume that we complete the deletion
+            removeContainerBasedGateway(api.getLabels().toArray()[0].toString(), api.getId());
```

Comment:
possible NPE?

Reply:
This is executed inside api.hasOwnGateway().
So having that value true means that the API has an autogenerated label always.
So will not give an NPE.